src/lib/OpenEXR/ImfAcesFile.h

```diff
@@ -150,7 +150,7 @@ class IMF_EXPORT_TYPE AcesOutputFile
     //------------------------------------------------
 
     IMF_EXPORT
-    void setFrameBuffer (const Rgba* base, size_t xStride, size_t yStride);
+    void setFrameBuffer (const Rgba* base, ptrdiff_t xStride, ptrdiff_t yStride);
```

Comment:
since an API change is involved, there'd be a version bump to OpenEXR required. Perhaps we should retain the existing signature in addition to introducing a new one, and have the size_t variant call the ptrdiff_t variant with appropriate casting to silence warnings.

Reply:
I think both gcc and clang (-Wall) would not produce warnings when passing `size_t` argument to `ptrdiff_t` parameter.

But these changes do break the ABI. However, OpenEXR usually has version number in its soname. So the binary capability would break after the next minor release anyway. Should I try to keep ABI capability in this PR?

Reply:
It strikes me that supporting flipping images on read is definitely a useful missing feature, but doing it by passing a negative stride is kind of a hack, rather than a clean friendly API feature. In particular, it's not obvious what the base pointer should be when the image dataWindow doesn't match the displayWindow.
One option would be to leave the constructor using size_t, but use ptrdiff_t internally, and add a new Make() static method for flipped images that createa a Slice with a negative stride and computes the correct base pointer for you.
That should also be a less significant ABI change than modifying the constructor.

Reply:
I'm writing a python extension. In my use case, I would definitely want to pass negative strides directly, because I could get negative strides from NumPy.

If we want to be more friendly to C++ users, instead of adding more `Make` overloads, how about adding `Slice::flipX()` and `Slice::flipY()` methods that negate the strides of an existing `Slice` and compute the base pointer?

Reply:
To my way of thinking, passing ptrdiff_t is less confusing at first blush than creating an unflipped datawindow. The displayWindow has to be an "increasing in memory, 0,0 upper left window". 

would it make sense to say displaywindow and datawindow have to be in harmony?

in other words, if xs < 0, then origin is right instead of left, if ys < 0, origin is bottom instead of top?

Reply:
If you mean the `base` pointer, then yes. the `base` pointer always points to the logical (0,0) pixel in the .exr file, but may not point to the first byte in the FrameBuffer Slice with negative stride. And this formula should always hold true:

https://github.com/AcademySoftwareFoundation/openexr/blob/55bdfd181200cc098ddd9f3919019250e1ba7ee6/src/lib/OpenEXR/ImfFrameBuffer.h#L49

`.header().dataWindow().min` should still less than or equal to `.header().dataWindow().max`